eventsources/sources/github/hook_util.go

```diff
@@ -1,32 +1,64 @@
 package github
 
 import (
+	"sort"
+
 	gh "github.com/google/go-github/v31/github"
 )
 
 // sliceEqual returns true if the two provided string slices are equal.
 func sliceEqual(first []string, second []string) bool {
-	if len(first) != len(second) {
+	if len(first) == 0 && len(second) == 0 {
+		return true
+	}
+	if len(first) == 0 || len(second) == 0 {
 		return false
 	}
 
-	if first == nil && second == nil {
+	uniqFirst := unique(first)
+	uniqSecond := unique(second)
+
+	sort.Strings(uniqFirst)
+	sort.Strings(uniqSecond)
+
+	// all_request
+	if contains(uniqFirst, "*") && contains(uniqSecond, "*") {
 		return true
 	}
 
-	if first == nil || second == nil {
+	if len(uniqFirst) != len(uniqSecond) {
 		return false
 	}
 
-	for index := range first {
-		if first[index] != second[index] {
+	for index := range uniqFirst {
+		if uniqFirst[index] != uniqSecond[index] {
 			return false
 		}
 	}
 
 	return true
 }
 
+func contains(sortedList []string, str string) bool {
+	i := sort.SearchStrings(sortedList, str)
+	return i < len(sortedList) && sortedList[i] == str
+}
+
+func unique(stringSlice []string) []string {
+	if len(stringSlice) == 0 {
```

Comment:
maybe some util for this some already?

Reply:
https://stackoverflow.com/questions/36000487/check-for-equality-on-slices-without-order